packages/block-library/src/pattern/recursion-detector.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * THIS MODULE IS INTENTIONALLY KEPT WITHIN THE PATTERN BLOCK'S SOURCE.
+ *
+ * This is because this approach for preventing infinite loops due to
+ * recursively rendering blocks is specific to the way that the `core/pattern`
+ * block behaves in the editor. Any other block types that deal with recursion
+ * SHOULD USE THE STANDARD METHOD for avoiding loops:
+ *
+ * @see https://github.com/WordPress/gutenberg/pull/31455
+ * @see packages/block-editor/src/components/recursion-provider/README.md
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useRegistry } from '@wordpress/data';
+
+const patternDependenciesRegistry = new WeakMap();
```

Comment:
With the introduction of the data registry stuff, this nomenclature is getting confusing.

Also, I ended up minimising what we store in the weak map, so instead of storing some kind of instance (e.g. `new PatternDependencies()`), we just store a bound function (`parsePatternDependencies.bind( null, new Map() )`). So it's even less of a "registry".

Suggestions? Examples:

```js
const validators = new WeakMap();
const patternParsers = new WeakMap();

// or simply...
const cache = new WeakMap();
```

Reply:
I'm the wrong person to ask about naming stuff